### DISCUSSION TITLE
hypergraphconv attentiion gradient = 0

### DISCUSSION THREAD

**Discussion by wuyiyang66** [pyg-team/pytorch_geometric#8351] [General]
I use pyg.nn.hypergraphConv to build a GCN model. Hypergraph attention is also used. When checking gradients, I found zero gradient of parameter gcn.att. I wonder why it happened.

Here is my code:

class HypGraph(nn.Module):
    def __init__(self, dim, gcn_hidden, gcn_out):
        super(HypGraph, self).__init__()
        self.dim = dim
        self.gcn_hidden = gcn_hidden
        self.gcn_out = gcn_out
        # self.embedding = nn.Embedding(29, self.dim, max_norm=True)
        self.gcn1 = gnn.HypergraphConv(
            self.dim,
            self.gcn_hidden,
            use_attention=True,
            heads=4,
            concat=True,
            dropout=0.1,
        )
        self.gcn2 = gnn.HypergraphConv(
            4 * self.gcn_hidden,
            self.gcn_out,
            use_attention=False,
            heads=1,
            concat=False,
            dropout=0.1,
        )

    def forward(self, gcn_in, ve):
        ef_list = []
        for idx in range(torch.max(ve[1, :]) + 1):
            ef_list.append(torch.mean(gcn_in[ve[0, :][ve[1, :] == idx]], dim=0))
        ef = torch.stack(ef_list)

        # ef = self.embedding(torch.arange(29).to(torch.device("cuda")))

        gcn_out = self.gcn1(x=gcn_in, hyperedge_index=ve, hyperedge_attr=ef)
        gcn_out = self.gcn2(x=gcn_out, hyperedge_index=ve, hyperedge_attr=ef)
        # gcn_out = gcn_in + gcn_out
        return gcn_out
